e2e_playwright/st_pills_test.py

```diff
@@ -0,0 +1,185 @@
+    wait_for_app_run(themed_app)
+    # click on second element to test multiselect
+    get_pill_button(pills, "🪢").click()
+    wait_for_app_run(themed_app)
+    text = get_markdown(themed_app, "Multi selection: \\['📝 Text', '🪢 Graphs'\\]")
+    expect(text).to_be_visible()
+
+    # click on same element to test unselect
+    get_pill_button(pills, "🪢").click()
+    wait_for_app_run(themed_app)
+    text = get_markdown(themed_app, "Multi selection: \\['📝 Text'\\]")
+    expect(text).to_be_visible()
+
+    # click on same element and take screenshot of multiple selected pills
+    get_pill_button(pills, "🪢").click()
+    # take away hover focus of button
+    themed_app.get_by_test_id("stApp").click(position={"x": 0, "y": 0})
+    wait_for_app_run(themed_app)
+    text = get_markdown(themed_app, "Multi selection: \\['📝 Text', '🪢 Graphs'\\]")
+    expect(text).to_be_visible()
+
+    assert_snapshot(pills, name="st_pills-multiselect")
+
+
+def test_click_single_icon_pill_and_take_snapshot(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test icon only pills (via format_func) and take a screenshot.
+
+    Click on same element to test unselect.
+    Click on two different elements to validate single select.
+    """
+
+    pills = get_button_group(themed_app, 1)
+
+    # the icon's span element has the respective text
+    # (e.g. :material/zoom_out_map: -> zoom_out_map)
+    get_pill_button(pills, "zoom_out_map").click()
+    text = get_markdown(themed_app, "Single selection: 3")
+    expect(text).to_be_visible()
+
+    # test unselect in single-select mode
+    get_pill_button(pills, "zoom_out_map").click()
+    text = get_markdown(themed_app, "Single selection: None")
+    expect(text).to_be_visible()
+
+    get_pill_button(pills, "zoom_in").click()
+    # take away hover focus of button
+    themed_app.get_by_test_id("stApp").click(position={"x": 0, "y": 0})
+    wait_for_app_run(themed_app)
+    text = get_markdown(themed_app, "Single selection: 1")
+    expect(text).to_be_visible()
+
+    assert_snapshot(pills, name="st_pills-singleselect_icon_only")
+
+
+def test_pass_default_selections(app: Page):
+    """Test that passed defaults are rendered correctly."""
+    text = get_markdown(app, "Multi selection: \\[\\]")
+    expect(text).to_be_visible()
+
+    click_checkbox(app, "Set default values")
+    text = get_markdown(
+        app, "Multi selection: \\['🧰 General widgets', '📊 Charts', '🧊 3D'\\]"
+    )
+    expect(text).to_be_visible()
+
+    click_checkbox(app, "Set default values")
+    text = get_markdown(app, "Multi selection: \\[\\]")
+    expect(text).to_be_visible()
+
+
+def test_selection_via_on_change_callback(app: Page):
+    """Test that the on_change callback is triggered when a pill is clicked."""
+    pills = get_button_group(app, 2)
+    get_pill_button(pills, "Air").click()
+    wait_for_app_run(app)
+    expect_markdown(app, "on_change selection: Air")
+
+
+def test_pills_are_disabled(app: Page):
```

Comment:
maybe also add a screenshot test here to make sure that the "disabled" styling aspects are also as expected

Reply:
great catch 👍